src/algebra/ring.lean

```diff
@@ -47,8 +49,77 @@ domain, integral_domain, nonzero_comm_semiring, nonzero_comm_ring, units
 universes u v w
 variable {α : Type u}
 
-section
-variable [semiring α]
+set_option default_priority 100 -- see Note [default priority]
+set_option old_structure_cmd true
+
+mk_simp_attribute field_simps "The simpset `field_simps` is used by the tactic `field_simp` to
+reduce an expression in a field to an expression of the form `n / d` where `n` and `d` are
+division-free."
+
+@[ancestor has_mul has_add]
+class distrib (α : Type u) extends has_mul α, has_add α :=
+(left_distrib : ∀ a b c : α, a * (b + c) = (a * b) + (a * c))
+(right_distrib : ∀ a b c : α, (a + b) * c = (a * c) + (b * c))
+
+lemma left_distrib [distrib α] (a b c : α) : a * (b + c) = a * b + a * c :=
+distrib.left_distrib a b c
+
+def mul_add := @left_distrib
```

Comment:
Do we want to keep all these aliases or just rename things?

Reply:
I'm planning a second refactor PR after this one to actually apply mathlib style stuff to the algebra files. This PR is just about reorganizing the dependencies and putting everything in order.